pkg/bootstrap/config.go

```diff
@@ -560,6 +562,8 @@ func GetNodeMetaData(options MetadataOptions) (*model.Node, error) {
 		return nil, err
 	}
 
+	meta = SetIstioVersion(meta)
```

Comment:
I'm not totally convinced by the approach of creating an additional function here...

I went down this route to separately test the logic, as `GetNodeMetaData` is a complex method and the related test covers a subset of the scenarios, and it seemed to me that adding only the additional IstioVersion scenarios to `TestGetNodeMetaData` made the test confusing, and trying to cover every scenarios was making the PR bigger and bigger. 

Thoughts?